src/components/custom-marker/custom-marker.component.tsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import { StyleSheet, Text, View } from "react-native";
+import { Marker } from "react-native-maps";
+import { Location } from "../../types/main";
+import { CONCORDIA_RED, BUILDING_MARKER_COLOR } from "../../constants/style";
+
+interface IProps {
+  location: Location;
+  onPress: () => void;
+  text: string;
+  markerType: "campus" | "building" | "poi";
+}
+let backgroundColor: string = CONCORDIA_RED;
+
+/**
+ * Custom component for rendering a marker with custom text and color
+ *
+ * @param location location for the marker
+ * @param onPress action to be done when marker is pressed
+ * @param text text of the marker
+ * @param markerType type of the marker
+ */
+const CustomMarker = ({ location, onPress, text, markerType }: IProps) => {
+  switch (markerType) {
+    case "campus":
+    case "poi":
+      backgroundColor = CONCORDIA_RED;
+      break;
+    case "building":
+      backgroundColor = BUILDING_MARKER_COLOR;
+      break;
+  }
+
+  return (
+    <Marker
+      testID={"marker"}
```

Comment:
Isn't this a code smell? Our source code has code specifically for testing

Reply:
Yes it is technically a test smell to have these ids that will only be used for testing. However, I discussed with @RGPosadas  it is the most reliable and easy way that enzyme provides to find elements for testing.

Reply:
It's specifically for Detox, not necessarily Enzyme (tho it would benefit Enzyme with finding props)
It's also a react native thing specifically for e2e testing
Sources: https://reactnative.dev/docs/view.html#testid